src/OrchardCore.Modules/OrchardCore.Email/Services/SmtpSettingsConfiguration.cs

```diff
@@ -12,41 +14,47 @@ public class SmtpSettingsConfiguration : IConfigureOptions<SmtpSettings>
         private readonly ISiteService _site;
         private readonly IDataProtectionProvider _dataProtectionProvider;
         private readonly ILogger<SmtpSettingsConfiguration> _logger;
+        private SmtpSettings _smtpSettings;
 
         public SmtpSettingsConfiguration(
+            IShellConfiguration shellConfiguration,
             ISiteService site,
             IDataProtectionProvider dataProtectionProvider,
             ILogger<SmtpSettingsConfiguration> logger)
         {
             _site = site;
             _dataProtectionProvider = dataProtectionProvider;
             _logger = logger;
+            _smtpSettings = shellConfiguration.GetSection("OrchardCore.Email").Get<SmtpSettings>();
+
         }
 
         public void Configure(SmtpSettings options)
         {
-            var settings = _site.GetSiteSettingsAsync()
-                .GetAwaiter().GetResult()
-                .As<SmtpSettings>();
+            if (_smtpSettings == null)
+            {
+                _smtpSettings = _site.GetSiteSettingsAsync()
+                    .GetAwaiter().GetResult().As<SmtpSettings>();
+            }
 
-            options.DefaultSender = settings.DefaultSender;
-            options.DeliveryMethod = settings.DeliveryMethod;
-            options.PickupDirectoryLocation = settings.PickupDirectoryLocation;
-            options.Host = settings.Host;
-            options.Port = settings.Port;
-            options.EncryptionMethod = settings.EncryptionMethod;
-            options.AutoSelectEncryption = settings.AutoSelectEncryption;
-            options.RequireCredentials = settings.RequireCredentials;
-            options.UseDefaultCredentials = settings.UseDefaultCredentials;
-            options.UserName = settings.UserName;
+            options.DefaultSender = _smtpSettings.DefaultSender;
+            options.DeliveryMethod = _smtpSettings.DeliveryMethod;
+            options.PickupDirectoryLocation = _smtpSettings.PickupDirectoryLocation;
+            options.Host = _smtpSettings.Host;
+            options.Port = _smtpSettings.Port;
+            options.EncryptionMethod = _smtpSettings.EncryptionMethod;
+            options.AutoSelectEncryption = _smtpSettings.AutoSelectEncryption;
+            options.RequireCredentials = _smtpSettings.RequireCredentials;
+            options.UseDefaultCredentials = _smtpSettings.UseDefaultCredentials;
+            options.UserName = _smtpSettings.UserName;
 
             // Decrypt the password
-            if (!String.IsNullOrWhiteSpace(settings.Password))
+            if (!String.IsNullOrWhiteSpace(_smtpSettings.Password))
             {
                 try
                 {
                     var protector = _dataProtectionProvider.CreateProtector(nameof(SmtpSettingsConfiguration));
-                    options.Password = protector.Unprotect(settings.Password);
+                    options.Password = protector.Unprotect(_smtpSettings.Password);
```

Comment:
This won't work from appsettings. The dataprotected value would have to be provided in appsettings, which means something would have to calculate it.

Plus this has now overridden (if supplied) all the email settings UI, with no indication to the admin that the settings they are supplying in the UI are not going to be used.

@hishamco I see what you are trying to do here, but I don't think it is the right thing.

I looked back on the code, and it looks like email config in appsettings was added accidentally (testing code) when workflows were merged

Reply:
I replied to you in #4656